packages/ingest/HttpProviderClient.js

```diff
@@ -65,13 +73,20 @@ class HttpProviderClient {
       c.on('fetchcomplete', (_, responseBuffer) => {
         const lines = responseBuffer.toString().trim().split('\n');
         lines.forEach((line) => {
-          const split = line.trim().split(pattern);
-          if (split.length === 3) {
-          // Some providers provide files with one number after the dot (".") ex (tmtdayacz8110_5.6)
-            if (split[1].match(/^(.*\.[\w\d]{1,4})\s*$/) !== null) {
-              const name = split[1].trimRight();
+          const trimmedLine = line.trim();
+          let match = matchLinksPattern.exec(trimmedLine);
+
+          while (match != null) {
```

Comment:
In theory you can do something like trimmedLine.match(matchLinksPattern) and get an array of strings to iterate over with a foreach/etc  if all you care about *is* the string and not other groupings.       It might read a bit easier, but not worth holding up the review on.